acdcli/api/content.py

```diff
@@ -61,57 +62,63 @@ class ContentMixin(object):
     """Implements content portion of the ACD API."""
 
     def create_folder(self, name: str, parent=None) -> dict:
-        body = {'kind': 'FOLDER', 'name': name}
-        if parent:
-            body['parents'] = [parent]
-        body_str = json.dumps(body)
+        while True:
+            body = {'kind': 'FOLDER', 'name': name}
+            if parent:
+                body['parents'] = [parent]
+            body_str = json.dumps(body)
 
-        acc_codes = [http.CREATED]
+            acc_codes = [http.CREATED]
 
-        r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            if r.status_code in RETRY_CODES: continue  # the fault lies not in our stars, but in amazon
```

Comment:
The example is that the amazon server always returns a 500 for every retry. My comment was meant to change the code in a way to be prepared for the exceptional case. We all make mistakes and therefore it is possible that the loop can happen in the future. To have a limit for retry would allow handling such situations gracefully.